### ISSUE TITLE
ViewModels request new data each time view is recreated

### ISSUE THREAD

**Issue by chris-feldman** [googlesamples/android-sunflower#24]
If PlantListFragment or GardenPlantingListFragment are recreated, their ViewModels make a new database request instead of returning the last value held by the livedata.

For example:

```
fun getGardenPlantings() = gardenPlantingRepository.getGardenPlantings()

fun getPlantAndGardenPlantings() = gardenPlantingRepository.getPlantAndGardenPlantings()
```

The Arch docs say this is a not to do this ([link](https://developer.android.com/topic/libraries/architecture/livedata#transform_livedata)):

```
    private LiveData<String> getPostalCode(String address) {
       // DON'T DO THIS
       return repository.getPostCode(address);
    }
```

> The UI component then needs to unregister from the previous LiveData object and register to the new instance each time it calls getPostalCode(). In addition, if the UI component is recreated, it triggers another call to the repository.getPostCode() method instead of using the previous call’s result.

The Arch components BasicSample shows a better way to implement this using MediatorLiveData ([link](https://github.com/googlesamples/android-architecture-components/blob/master/BasicSample/app/src/main/java/com/example/android/persistence/viewmodel/ProductListViewModel.java))

When I implemented this myself, using the MediatorLiveData also fixed #10 
https://github.com/chris-feldman/android-sunflower/blob/master/app/src/main/java/com/google/samples/apps/sunflower/viewmodels/PlantListViewModel.kt

**Comment by tiembo** [comment]
Thanks for the detailed write-up!  Would you be interested in opening a pull request to update the two list ViewModels?

**Comment by chris-feldman** [comment]
Sure, I'll see if I can get one together today 👍

**Comment by sagar-viradiya** [comment]
I think better solution is to use new fragment life cycle while observing live data.

Read this

https://medium.com/@BladeCoder/architecture-components-pitfalls-part-1-9300dd969808

check note at end of post

We need to update support lib as new view life cycle is there in 28.0.0

**Comment by sagar-viradiya** [comment]
Yeah sorry my bad. Thing is every time you switch fragment from GardenPlantListing to PlantListing and vice versa onCreateView will get call and method in view model also. Which will end up in querying database multiple times. Your PR will fix this issue.

However there is one issue of multiple observation on livedata. So each time you switch fragment new live data observer is being registered. This will lead to multiple callbacks for livedata observer and because of this multiple notifyDataSetChanged() in adapter.

I have figured out this issue. @tiembo Should I create new issue for this ?

**Comment by tiembo** [comment]
Hi @sagar-viradiya - yes, please open a new issue and we can discuss the livedata changes there.

**Comment by tiembo** [comment]
Closing issue; fixed via #33
